### PR TITLE
docs: Add libvips-dev as a gatsby dependancy on debian/ubuntu

### LINKED ISSUE #12101: gatsbyjs.org build assumes native dependency `libvips-dev`
## Description

`yarn run develop` for gatsbyjs.org (`www/`) fails with following errors on a fresh ubuntu 18.04 setup:
```
error Failed to process image /path/to/gatsby/www/.cache/gatsby-source-filesystem/31986f7e77b390d77efe1d58936176a0/db0d7d37f2039e29db5e4611a2a19392.png
  Error: vipspng: libpng read error
  vipspng: libpng read error
  vips2png: unable to write to buffer
```

Installing `libvips-dev`  fixed the issue; maybe we should note the native dependency in the readme? 
Although it's fairly easy to figure it out.

### PR DESCRIPTION
**Pull request by chiamakaikeanyi**
## Description
 Running `yarn run develop` for gatsbyjs.org (`www/`) fails with the following errors on a fresh ubuntu 18.04 setup:

```
error Failed to process image /path/to/gatsby/www/.cache/gatsby-source-filesystem/31986f7e77b390d77efe1d58936176a0/db0d7d37f2039e29db5e4611a2a19392.png
  Error: vipspng: libpng read error
  vipspng: libpng read error
  vips2png: unable to write to buffer
```
Installing `libvips-dev` fixed the issue. 

This PR addresses the issue of informing users of the solution to a possible error on Linux

## Related Issues
 Fixes #12101

### CODE REVIEW

**Comment by yogeshkotadiya** [comment]
Just to clarify `libvips-dev` is a Debian package. 
Different distributions have a different name for vips library.
[https://pkgs.org/download/vips](https://pkgs.org/download/vips)
And also this should be on [Gatsby-on-linux](https://www.gatsbyjs.org/docs/gatsby-on-linux/) page (correct me if I'm wrong).

**Comment by chiamakaikeanyi** [comment]
@yogeshkotadiya, I can't seem to find it on the page. You can have a look.

**Comment by pieh** [comment]
I think this should be primarily on `Gatsby on Linux` page, because this is not 
gatsbyjs.org specific error (it can happen on any site that uses `sharp` plugins).

The [`gatsby-on-linux`](https://www.gatsbyjs.org/docs/gatsby-on-linux/) page is super weird - because it has Linux part straight mentioned as TO-DO and all instructions are for using Linux on Windows 😂

File: docs/contributing/blog-and-website-contributions.md [hunk 1]
```diff
@@ -58,6 +58,7 @@ If you want to make changes, improvements, or add new functionality to the websi
 
 - Clone [the Gatsby repo](https://github.com/gatsbyjs/gatsby/) and navigate to `/www`
 - Run `yarn` to install all of the website's dependencies.
+- If you are experiencing issues on a Linux machine, run `sudo apt install libvips-dev`, to install a native dependency. You can also reference [Gatsby guide on Linux](/docs/gatsby-on-linux/) for other Linux-specific requirements.
```
Comment (sidharthachatterjee) [inline docs/contributing/blog-and-website-contributions.md]: Maybe this works better as a foot note as opposed to a point in the regular instructions?

Comment (chiamakaikeanyi) [reply docs/contributing/blog-and-website-contributions.md]: Do you mean that I should use link

- Run `yarn` to install all of the website's dependencies.<sup id="linuxdependency">[1](#libvips)</sup>

and add it at the footer like the sample below?

<a id="libvips"><b>1</b></a>: If you are experiencing issues on a Linux machine, run `sudo apt install libvips-dev`, to install a native dependency. You can also reference [Gatsby guide on Linux](/docs/gatsby-on-linux/) for other Linux-specific requirements. [↩](#linuxdependency)

Comment (marcysutton) [reply docs/contributing/blog-and-website-contributions.md]: I think you could put this item on its own line and start it with `> Note:`
That would be consistent with other notes in the docs.

Comment (chiamakaikeanyi) [reply docs/contributing/blog-and-website-contributions.md]: @marcysutton Can you help review my PR. It has been updated.

File: docs/contributing/blog-and-website-contributions.md [hunk 2]
```diff
@@ -60,6 +60,8 @@ If you want to make changes, improvements, or add new functionality to the websi
 - Run `yarn` to install all of the website's dependencies.
 - Run `npm run develop` to preview the site at `http://localhost:8000/`.
 
+Note: If you are experiencing issues on a Linux machine, run `sudo apt install libvips-dev`, to install a native dependency. You can also reference [Gatsby guide on Linux](/docs/gatsby-on-linux/) for other Linux-specific requirements.
```
Comment (yogeshkotadiya) [inline docs/contributing/blog-and-website-contributions.md]: You forgot to add `>`
```suggestion
> Note: If you are experiencing issues on a Linux machine, run `sudo apt install libvips-dev`, to install a native dependency. You can also reference [Gatsby guide on Linux](/docs/gatsby-on-linux/) for other Linux-specific requirements.
```

Comment (chiamakaikeanyi) [reply docs/contributing/blog-and-website-contributions.md]: @marcysutton suggested that I should make it consistent with others. `>` is not included in others

Comment (chiamakaikeanyi) [reply docs/contributing/blog-and-website-contributions.md]: Thanks.

File: docs/contributing/blog-and-website-contributions.md (hunk 1 above)
Comment (polarathene) [reply docs/contributing/blog-and-website-contributions.md]: The command given is only valid if you're using `apt` as the package manager(Ubuntu/Debian), on other distros it won't work.

Sharp pulls in linux binaries that should work without needing this(unless you have musl like on Alpine Linux), iirc you need to be using a compatible version of Node for the binaries though(v8 and v10 I think, see the docs for sharp, it has a section about this). If no compatible binary is available, sharp then attempts to compile. **EDIT:** [Here's those docs](https://sharp.dimens.io/en/stable/install/), it's better to link there for troubleshooting.

**Pull request merged**